debezium-embedded/src/main/java/io/debezium/embedded/Transformations.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.embedded;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.kafka.connect.source.SourceRecord;
+import org.apache.kafka.connect.transforms.Transformation;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.debezium.DebeziumException;
+import io.debezium.config.Configuration;
+
+/**
+ * Composite class representing transformation chain.
+ *
+ * @author Jiri Pechanec
+ *
+ */
+public class Transformations implements Closeable {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(Transformations.class);
+
+    private static final String TYPE_SUFFIX = ".type";
+
+    private final Configuration config;
+    private final List<Transformation<SourceRecord>> transforms = new ArrayList<>();
+
+    public Transformations(Configuration config) {
+        this.config = config;
+        final String transformationList = config.getString(EmbeddedEngine.TRANSFORMS);
+        if (transformationList == null) {
+            return;
+        }
+        for (String transfName : transformationList.split(",")) {
+            transfName = transfName.trim();
+            final Transformation<SourceRecord> transformation = getTransformation(transfName);
+            transformation.configure(config.subset(transformationConfigNamespace(transfName), true).asMap());
+            transforms.add(transformation);
+        }
+    }
+
+    private String transformationConfigNamespace(final String name) {
+        return EmbeddedEngine.TRANSFORMS.name() + "." + name;
+    }
+
+    @SuppressWarnings("unchecked")
+    private Transformation<SourceRecord> getTransformation(String name) {
+        try {
+            final Transformation<SourceRecord> transformation = config.getInstance(EmbeddedEngine.TRANSFORMS.name() + "." + name + TYPE_SUFFIX, Transformation.class);
+            if (transformation == null) {
+                throw new DebeziumException("Cannot instatiate transformation '" + name + "'");
+            }
+            return transformation;
+        }
+        catch (Exception e) {
+            throw new DebeziumException("Error while instantiating transformation '" + name + "'");
```

Comment:
You might end up with a wrapped `DebeziumException` here.

Reply:
Done